File: rules_sonatype/src/main/scala/org/jetbrains/bazel/sonatype/SonatypeCoordinates.scala
package org.jetbrains.bazel.sonatype

case class SonatypeCoordinates(sonatypeGroupId: String, sonatypeArtifactId: String, sonatypeVersion: String)

object SonatypeCoordinates {
  def apply(sonatypeCoordinates: String): SonatypeCoordinates = {
    val parts = sonatypeCoordinates.split(":")
    if (parts.size != 3)
      throw new IllegalArgumentException("Coordinates must be a triplet, got: " + sonatypeCoordinates)

    new SonatypeCoordinates(parts(0), parts(1), parts(2))
  }
}

File: sonatype_kt/src/main/kotlin/org/jetbrains/bazel/sonatype/SonatypeCoordinates.kt
package org.jetbrains.bazel.sonatype

data class SonatypeCoordinates(
  val sonatypeGroupId: String,
  val sonatypeArtifactId: String,
  val sonatypeVersion: String
) {
  companion object {
    fun fromString(coordinates: String): SonatypeCoordinates {
      val parts = coordinates.split(":")
      if (parts.size != 3) throw IllegalArgumentException("Coordinates must be a triplet, got: $coordinates")
      return SonatypeCoordinates(parts[0], parts[1], parts[2])
    }
  }
}
